src/WROVER_KIT_LCD.cpp

```diff
@@ -114,6 +114,50 @@ uint32_t WROVER_KIT_LCD::readId() {
     return r;
 }
 
+
+uint16_t WROVER_KIT_LCD::readPixels(uint16_t *colors, uint32_t len) {
+    uint8_t f;
+    uint16_t ret = len;
+    writeCommand(0xD9);
+    SPI.write(0x10);
+    writeCommand(WROVER_RAMRD);
+    f = SPI.transfer(0); // dummy read
+    f = SPI.transfer(0); // dummy read
+    while(len--) {    
+      uint8_t r = SPI.transfer(0);
+      uint8_t g = SPI.transfer(0);
+      uint8_t b = SPI.transfer(0);
+      *colors++ = color565(r, g, b);
+    }
+    return ret;
+}
+
+
+uint16_t WROVER_KIT_LCD::readPixel(int16_t x, int16_t y) {
```

Comment:
this method is redundant, or add `uint16_t readPixel()` as a variant also

Reply:
Using `readPixel()` alone without frequency change will result in a corrupted palette. In my opinion this is what it should look like to return non-corrupted colors, assuming a `setAddrWindow()` call occured earlier:

```c
uint16_t WROVER_KIT_LCD::readPixel() {
    uint32_t freq = _freq;
    if(_freq > 16000000){
        _freq = 16000000;
    }
    uint16_t buf[1];
    startWrite(); // begin transaction
    readPixels(buf, 1);
    endWrite(); // end transaction
    _freq = freq;
    return buf[0];
}
```